plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -640,19 +640,80 @@ public function get_total_shipping_refunded( $order ) {
 		return 0;
 	}
 
+	/**
+	 * Finds an Order ID based on an order key.
+	 *
+	 * @param string $order_key An order key has generated by.
+	 * @return int The ID of an order, or 0 if the order could not be found
+	 */
 	public function get_order_id_by_order_key( $order_key ) {
-		// TODO: Implement get_order_id_by_order_key() method.
-		return 0;
+		global $wpdb;
+
+		$orders_table = self::get_orders_table_name();
+		$op_table     = self::get_operational_data_table_name();
+
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		return $wpdb->get_var(
+			$wpdb->prepare(
+				"SELECT {$orders_table}.id FROM {$orders_table}
+				INNER JOIN {$op_table} ON {$op_table}.order_id = {$orders_table}.id
+				WHERE {$op_table}.order_key = %s",
+				$order_key
+			)
+		);
+		// phpcs:enable
 	}
 
+	/**
+	 * Return count of orders with a specific status.
+	 *
+	 * @param  string $status Order status. Function wc_get_order_statuses() returns a list of valid statuses.
+	 * @return int
+	 */
 	public function get_order_count( $status ) {
-		// TODO: Implement get_order_count() method.
-		return 0;
+		global $wpdb;
+
+		$orders_table = self::get_orders_table_name();
+
+		return absint( $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM {$orders_table} WHERE type = %s AND status = %s", 'shop_order', $status ) ) ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+	}
+
+	/**
+	 * Get all orders matching the passed in args.
+	 *
+	 * @deprecated 3.1.0 - Use {@see wc_get_orders} instead.
+	 * @param  array $args List of args passed to wc_get_orders().
+	 * @return array|object
+	 */
+	public function get_orders( $args = array() ) {
+		wc_deprecated_function( __FUNCTION__, '3.1.0', 'Use wc_get_orders instead.' );
+		return wc_get_orders( $args );
 	}
 
+	/**
+	 * Get unpaid orders after a certain date,
```

Comment:
```suggestion
	 * Get unpaid orders last updated before the specified date.
```

So I know the current description matches what we have in the interface and in the CPT data-store ... but unless I've been staring at the code too long (which is possible ¯\\\_(ツ)_/¯) aren't we querying for orders where `date_updated_gmt` is **less than** the provided timestamp?

If I'm just confused, we should probably still amend and kill the trailing `,` and replace with a `.`.

Reply:
I'm not entirely sure if this is an error or not. I guess the "spirit" of the method was that if an order is unpaid and was last updated before $date, it can be considered "unpaid" from that $date on. Meanwhile, unpaid orders last updated >= $date could be "in the process of being paid" 🤷.

Anyways, I do think your suggestion makes the purpose of the method much clearer.